src/studio/src/designer/backend/Controllers/UIEditorController.cs

```diff
@@ -93,6 +96,43 @@ public ActionResult GetTextResources(string org, string app, string id)
             return Content(result);
         }
 
+        /// <summary>
+        /// Add text resources to existing resource documents
+        [HttpPost]
+        public ActionResult AddTextResources(string org, string app, [FromBody] List<TextResource> textResources)
+        {
+            foreach (TextResource textResource in textResources)
```

Comment:
Consider moving the logic to the repository

Reply:
Agreed, and done :)